sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -656,6 +656,25 @@ object SQLConf {
       .intConf
       .createWithDefault(10000)
 
+  val PARALLEL_GET_GLOBBED_PATH_THRESHOLD =
+    buildConf("spark.sql.sources.parallelGetGlobbedPath.threshold")
+      .doc("The maximum number of subfiles or directories allowed after a globbed path " +
+        "expansion.")
+      .intConf
+      .checkValue(threshold => threshold >= 0, "The maximum number of subfiles or directories " +
```

Comment:
`internal`?

Reply:
Maybe we should keep this public? Because the parallel only opened when the thread number > 0.